### PR TITLE
Supply: Receive/Dispense question types

### PR DESCRIPTION
**Pull request by millerdev**
https://trello.com/c/HYmHNRua

cc @kaapstorm

### CODE REVIEW

File: src/commtrack.js [hunk 1]
```diff
@@ -263,61 +279,85 @@ define([
...
+        dispenseMugOptions = util.extend(transferMugOptions, {
+            typeName: 'Dispense',
+            icon: 'fcc fcc-fd-door',
+            spec: {
+                src: {
+                    validationFunc: function (mug) {
+                        if (mug.p.src.value) {
+                            return 'pass';
+                        }
+                        return 'Receive must have a Source Case.';
```
Comment (czue) [inline src/commtrack.js]: I think this should say "dispense"

File: src/commtrack.js [hunk 2]
```diff
@@ -263,61 +279,85 @@ define([
...
+                    },
+                },
+                dest: { presence: "notallowed" },
```
Comment (czue) [inline src/commtrack.js]: assume this hides it from the UI?

File: src/commtrack.js [hunk 3]
```diff
@@ -263,61 +279,85 @@ define([
...
+            types.normal.Transfer = transferMugOptions;
+            types.normal.Dispense = dispenseMugOptions;
+            types.normal.Receive = receiveMugOptions;
             return types;
         },
+        parseDataElement: function (form, el, parentMug, role) {
+            var tag = el.nodeName;
+            if (!role && (tag === "transfer" || tag === "balance") &&
+                    $(el).attr("xmlns") === LEDGER_XMLNS) {
+                if (tag === "transfer") {
```
Comment (czue) [inline src/commtrack.js]: out of curiosity is possible to change between these question types or you would have to manually migrate all the properties over if you wanted to add a src/dest to a single-sided transaction?

**Comment by czue** [comment]
looked good overall but I think @emord should probably do a full review

**Comment by czue** [comment]
thanks for banging this out!

File: src/commtrack.js (hunk 2 above)
Comment (millerdev) [inline src/commtrack.js]: correct

File: src/commtrack.js (hunk 3 above)
Comment (millerdev) [inline src/commtrack.js]: you can. I added that in d57ea083c026656b2b9e3ac0ee1c07449840c772

Comment (millerdev) [inline src/commtrack.js]: there are even tests for it [here](https://github.com/dimagi/Vellum/commit/d57ea083c026656b2b9e3ac0ee1c07449840c772#diff-ff401b6488cc00a539c9523227c94c15R314) and [here](https://github.com/dimagi/Vellum/commit/d57ea083c026656b2b9e3ac0ee1c07449840c772#diff-905dcd5f60ed0d9e12b350401d356a0aR416)

**Comment by millerdev** [comment]
Possibly adding/changing icons yet

File: src/commtrack.js [hunk 4]
```diff
@@ -89,21 +100,11 @@ define([
...
-                    ],
-                }),
-                logicSection
-            ]
+            Transfer: [basicSection, logicSection],
+            Dispense: [basicSection, logicSection],
+            Receive: [basicSection, logicSection],
         },
-        baseTransactionOptions = {
+        baseTransactionOptions = util.extend(mugs.defaultOptions, {
```
Comment (emord) [inline src/commtrack.js]: Side note: Maybe we could change this to `util.deepExtend` so as to not confuse it with `_.extend`

Comment (millerdev) [inline src/commtrack.js]: good idea

**Pull request merged**